EIPS/eip-1238.md

```diff
@@ -0,0 +1,213 @@
+- Implementers MUST allow token recipients to burn any token they receive.
+- Implementers MUST only allow tokens to be minted with the consent of the recipient.
+- Implementers MAY enable token issuers to burn the tokens they issued.
+
+**Smart contracts MUST implement all of the functions in the `ERC1238TokenReceiver` interface to accept tokens being minted to them.**
+
+The **URI Storage extension** is OPTIONAL for smart contracts that implement this EIP. This allows your contract to associate a unique URI for each token id.
+
+```solidity
+interface IERC1238URIStorage is IERC1238 {
+    /**
+     * @dev Emitted when the URI for token type `id` changes to `value`, if it is a non-programmatic URI.
+     */
+    event URI(uint256 indexed id, string value);
+
+    /**
+     * @dev Returns the Uniform Resource Identifier (URI) for `id` token.
+     */
+    function tokenURI(uint256 id) external view returns (string memory);
+}
+```
+
+The **Expirable extension** is OPTIONAL for smart contracts that implement this EIP. This allows your contract to associate a unique URI for each token id.
+
+```solidity
+interface IERC1238Expirable is IERC1238 {
+    /**
+     * @dev Returns the expiry date for tokens with a given `id`.
+     */
+    function expiryDate(uint256 id) external view returns (uint256);
+
+    /**
+     * @dev Returns whether tokens are expired by comparing their expiry date with `block.timestamp`.
+     */
+    function isExpired(uint256 id) external view returns (bool);
+
+    /**
+     * @dev Sets the expiry date for the tokens with id `id`.
+     */
+    function setExpiryDate(uint256 id, uint256 date) external;
+
+    /**
+     * @dev [Batched] version of {setExpiryDate}.
+     */
+    function setBatchExpiryDates(uint256[] memory ids, uint256[] memory dates) external;
+}
+```
+
+The **Holdable extension** is OPTIONAL for smart contracts that implement this EIP. This enables token owners to let their tokens be held by another address or to stake them in a smart contract.
+
+```solidity
+interface IERC1238Holdable is IERC1238 {
+    /**
+     * @dev Event emitted when `from` entrusts `to` with `amount` of tokens with token `id`.
+     */
+    event Entrust(address from, address to, uint256 indexed id, uint256 amount);
+
+    /**
+     * @dev Event emitted when tokens are burnt and the holder fails to acknowledge the burn.
+     */
+    event BurnAcknowledgmentFailed(address holder, address burner, address from, uint256 indexed id, uint256 amount);
+
+    /**
+     * @dev Returns the balance of a token holder for a given `id`.
+     */
+    function heldBalance(address holder, uint256 id) external view returns (uint256);
+
+    /**
+     * @dev Lets sender entrusts `to` with `amount`
+     * of tokens which gets transferred between their respective balances
+     * of tokens held.
+     */
+    function entrust(
+        address to,
+        uint256 id,
+        uint256 amount
+    ) external;
+}
+```
```

Comment:
This holdable extension needs a lot more explanation. You should define the concepts, the expected flow, etc.

Reply:
I added more explanations in https://github.com/ethereum/EIPs/pull/5617/commits/a17e69dd6a6a2301d53ba01fdcc93f694aaa95c7